tests/Doctrine/Tests/ORM/Query/QueryTest.php

```diff
@@ -303,4 +304,65 @@ public function testSelectFromSubquery()
         $this->expectExceptionMessage('Subquery');
         $query->getSQL();
     }
+
+    /**
+     * @group #6699
+     */
+    public function testGetParameterTypeJuggling() : void
+    {
+        $query = $this->_em->createQuery('select u from ' . CmsUser::class . ' u where u.id = ?0');
+
+        $query->setParameter(0, 0);
+
+        $this->assertSame(0, $query->getParameter(0)->getValue());
+        $this->assertSame(0, $query->getParameter('0')->getValue());
+    }
+
+    /**
+     * @group #6699
+     */
+    public function testSetParameterWithNameZeroIsNotOverridden() : void
+    {
+        $query = $this->_em->createQuery('select u from ' . CmsUser::class . ' u where u.id != ?0 and u.username = :name');
+
+        $query->setParameter(0, 0);
+        $query->setParameter('name', 'Doctrine');
+
+        $this->assertSame(2, $query->getParameters()->count());
+        $this->assertSame(0, $query->getParameter('0')->getValue());
+        $this->assertSame('Doctrine', $query->getParameter('name')->getValue());
+    }
+
+    /**
+     * @group #6699
+     */
+    public function testSetParameterWithNameZeroDoesNotOverrideAnotherParameter() : void
+    {
+        $query = $this->_em->createQuery('select u from ' . CmsUser::class . ' u where u.id != ?0 and u.username = :name');
+
+        $query->setParameter('name', 'Doctrine');
+        $query->setParameter(0, 0);
+
+        $this->assertSame(2, $query->getParameters()->count());
+        $this->assertSame(0, $query->getParameter(0)->getValue());
+        $this->assertSame('Doctrine', $query->getParameter('name')->getValue());
+    }
+
+    /**
+     * @group #6699
```

Comment:
We're using the ticket id without the `#` everywhere else, we should keep it consistent here too